_posts/2021-12-13-why-we-should-rewrite-enzymejs-to-rtl-and-how-to-do-that.md

```diff
@@ -0,0 +1,325 @@
+   directly to the internal component. While Enzyme uses shallow rendering (or sometimes deep rendering) and 
+   encourages us to test the components instance using their state and props. Enzyme also allows full rendering of DOM 
+   but it is not the basic assumption of the library’s operation.
+2. **Bundle size** 
+   
+   Maybe this is not as significant as the first point, but it is worth paying attention to bundle size also. RTL has 
+   a two times smaller bundle.
+3. **Age**  
+   
+   Enzyme is 6 years old, RTL is 3 years old
+
+## Comparison of downloads
+
+Below we can see a comparison of the number of downloads of both libraries in the last 5 years. Between August and 
+September 2020, the number of RTL downloads exceeded the number of Enzyme downloads.
+
+![Enzyme and RTL dowloands graph](/img/articles/2021-12-13-why-we-should-rewrite-enzymejs-to-rtl-and-how-to-do-that/06.png)
+###### Source [npm trends](https://www.npmtrends.com/@testing-library/react-vs-enzyme)
+
+In the case of RTL, it is clear that the number of downloads is still growing strongly. On the other hand, it can 
+be suspected that the highest peak of Enzyme downloads is already behind it.
+
+###### Reasons to change testing library
+
+Well, let’s be honest, Enzyme.js is a powerful library which helps us test our React Components in an easier way for a 
+long time. But maybe.. too long? The first official release was in December 2015. Do not get me wrong, I don’t want to 
+say that everything old is bad. But the frontend world is changing constantly, just as the approach to testing it. 
+I know that the Enzyme.js library hasn’t been deprecated yet, but there are a lot of reasons to switch your test library 
+right now. **If you haven’t read it I highly recommend you to take a look at this 
+[Piotr Staniów article](https://www.piotrstaniow.pl/goodbye-enzyme)** (it takes about 10 minutes to read). 
+
+The main reasons for me to think about slowly rewriting your tests and switch to another library from this article are: 
+- one maintains a developer, 
+- not keeping up with React changes. 
+
+I think these are big blockers in using this tool on.
+
+## Case study of rewriting
+
+Below I will present my own example created for the purposes of this article. It should present the minimal effort to 
+rewrite one library to another.
+
+### Component: 
+
+```js
+import React, { useState } from 'react';
+
+export const Test = ({ defaultCounter = 0, header = 'Default header' }) => {
+    const [counter, setCounter] = useState(defaultCounter)
+    const increment = () => setCounter(counter + 1)
+    const decrement = () => setCounter(counter - 1)
+
+    return (
+        <div>
+            <h1>{header}</h1>
+            <p className='counter'>{counter}</p>
+            <button onClick={increment}>Increment counter</button>
+            <button onClick={decrement}>Decrement counter</button>
+        </div>
+    )
+};
+
+export default Test;
```

Comment:
I see the named export is used in tests below, so the default one is not necessary